src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/CoreCLR/WebProxy.cs

```diff
@@ -18,6 +18,34 @@ internal WebProxy(Uri address)
             _proxyAddress = address;
         }
 
+        public override bool Equals(object obj)
+        {
+            if (obj == null || GetType() != obj.GetType())
+            {
+                return false;
+            }
+            
+            return Equals(obj as WebProxy);
+        }
+
+        public override int GetHashCode()
+        {
+            return (this._proxyAddress.GetHashCode() * 17) + this._credentials.GetHashCode();
+        }
+
+        public bool Equals(WebProxy other)
+        {
+            if (other is null)
+            {
+                return false;
+            }
+
+            // _proxyAddress cannot be null as it is set in the constructor            
+            return other._credentials == _credentials
```

Comment:
```suggestion
            return GetHashCode() == other.GetHashCode() && BypassProxyOnLocal == other.BypassProxyOnLocal;
```

Reply:
I would have thought even with comparing hashcode as a fast fail we should still make sure all other properties match for this to be considered equal.

```c#
          return GetHashCode() == other.GetHashCode() 
                && other._credentials == _credentials
                && _proxyAddress.Equals(other._proxyAddress)
                && BypassProxyOnLocal == other.BypassProxyOnLocal;
```

Reply:
> > Maybe you could do something like this for WebRequestSession.cs (it's just an example I think it can be done better)
> > ```cs
> >         internal int MaxAutomaticRedirections
> >         {
> >             get => _maxAutomaticRedirections;
> >             set => _maxAutomaticRedirections = SetVariable(typeof(int), _maxAutomaticRedirections, value);
> >         }
> > 
> >         internal dynamic SetVariable(Type type, object oldValue, object newValue)
> >         {
> >             dynamic convertedOldValue = Convert.ChangeType(oldValue, type);
> >             dynamic convertedNewValue = Convert.ChangeType(newValue, type);
> >             if (convertedOldValue != convertedNewValue)
> >             {
> >                 _changed = true;
> >                 return convertedNewValue;
> >             }
> >             else
> >             {
> >                 return convertedOldValue;
> >             }
> >         }
> > ```
> 
> I used a couple of generics to do something similar. I'm not a fan of using dynamic since it turns off lots of type checking.
> 
> ```cs
>    private void SetClassVar<T>(ref T oldValue, T newValue) where T : class
>         {
>             if (oldValue == newValue)
>             {
>                 return;
>             }
>             _changed = true;
>             oldValue = newValue;
>         }
> 
>         private void SetStructVar<T>(ref T oldValue, T newValue) where T : struct
>         {
>             if (oldValue.Equals(newValue))
>             {
>                 return;
>             }
>             _changed = true;
>             oldValue = newValue;
>         }
> ```
> 
> Let me know if there's a better way. I wasn't able to use a single generic for both structs and classes but I'm not an expert in this area and someone else may have a better suggestion.

I like your solution 👍